### PR TITLE
Fixes the node start IPC connection

### PR DESCRIPTION
**Pull request by frozeman**
This PR is adding a loop to test an IPC connection. When a connection timeout is set, it tries to connect every 200ms to the connection, if not it fails.

This speeds up the connection process, as we don't have to wait arbitrary 4 seconds anymore (which weren't enough for `eth`)

@hiddentao This PR works, but if you set a wrong IPC path it doesn't quit gracefully, instead i delivers an uncaught exception. Could you please look into this.
Simply let the `eth` node start (Change the `node` files content in `~/Library/App..Support/Mist/node` to `eth`) 
Before change the pic path in the `getIPCPath.js` to some other path.

You will see it fails, but not gracefully, as the error seems to travers through and doesn't get catched.

This error is created in the socket.js line 84

```js
[2016-06-16 16:44:28.827] [INFO] Sockets/node-ipc - Connect to {"path":"/Users/frozeman/Library/Ethereum/gethdddd.ipc"}
[2016-06-16 16:44:38.965] [ERROR] Sockets/node-ipc - { Error: connect ENOENT /Users/frozeman/Library/Ethereum/gethdddd.ipc
    at Object.exports._errnoException (util.js:949:11)
    at exports._exceptionWithHostPort (util.js:972:20)
    at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1080:14)
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'connect',
  address: '/Users/frozeman/Library/Ethereum/gethdddd.ipc' }
Error: connect ENOENT /Users/frozeman/Library/Ethereum/gethdddd.ipc
    at Object.exports._errnoException (util.js:949:11)
    at exports._exceptionWithHostPort (util.js:972:20)
    at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1080:14)
[2016-06-16 16:44:38.967] [ERROR] main - UNCAUGHT EXCEPTION { Error: connect ENOENT /Users/frozeman/Library/Ethereum/gethdddd.ipc
    at Object.exports._errnoException (util.js:949:11)
    at exports._exceptionWithHostPort (util.js:972:20)
    at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1080:14)
  code: 'ENOENT',
  errno: 'ENOENT',
  syscall: 'connect',
  address: '/Users/frozeman/Library/Ethereum/gethdddd.ipc' }
Error: connect ENOENT /Users/frozeman/Library/Ethereum/gethdddd.ipc
    at Object.exports._errnoException (util.js:949:11)
    at exports._exceptionWithHostPort (util.js:972:20)
    at PipeConnectWrap.afterConnect [as oncomplete] (net.js:1080:14)
```

### CODE REVIEW

File: modules/ethereumNode.js [hunk 1]
```diff
@@ -289,8 +289,8 @@ class EthereumNode extends EventEmitter {
                 this._saveUserData('network', this._network);
 
                 return this._socket.connect({ path: ipcPath }, {
-                    timeout: 30000 /* 30s */
-                })  
+                        timeout: 10000 /* 30s */
```
Comment (frozeman) [inline modules/ethereumNode.js]: We need to increase this to 30s again after its fixed

**Comment by hiddentao** [comment]
I will refactor the interval timer loop.

**Comment by hiddentao** [comment]
Changes:

* Commented out the master password window and logic, eth now just works
* Refactored the socket connection code, so that it waits for a result before re-connecting
* Now fails gracefully

**Comment by hiddentao** [comment]
@frozeman let me know what you think.

**Comment by hiddentao** [comment]
Also noticed that the splash screen is showing state download info, but for Eth there isn't any. This needs to be removed before this PR is complete.

**Comment by hiddentao** [comment]
Only pending issue - should we just remove the master password logic from the node startup?

**Comment by hiddentao** [comment]
Final changes made, ready for review.

**Comment by alexvandesande** [comment]
What's the status of this? Is it going in 0.7.6?

**Comment by hiddentao** [comment]
This is done and ready for review. It should definitely go into 0.7.6.

**Comment by hiddentao** [comment]
To test eth startup:

```shell
$ electron . --loglevel debug --node eth --network test
```

**Comment by frozeman** [comment]
we should remove the master password logic. And provide a script for people to get rid of it

**Comment by hiddentao** [comment]
It's removed.

**Comment by luclu** [comment]
Seems to work well - under windows 7 and OSX.
Test starting, switching networks/nodes, detaching.
**But** I needed to remove eth's config folder.
﻿﻿

**Pull request merged**